custom/icds_reports/utils/aggregation_helpers/monolith/awc_location.py

```diff
@@ -1,45 +1,107 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
-from six.moves import map
+import io
+import json
 
-from corehq.apps.userreports.models import StaticDataSourceConfiguration, get_datasource_config
+import csv342 as csv
+from six.moves import map, range
+
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.userreports.models import (
+    StaticDataSourceConfiguration,
+    get_datasource_config,
+)
 from corehq.apps.userreports.util import get_table_name
 from custom.icds_reports.const import AWC_LOCATION_TABLE_ID, AWW_USER_TABLE_ID
 from custom.icds_reports.utils.aggregation_helpers.monolith.base import BaseICDSAggregationHelper
-from six.moves import range
 
 
 class LocationAggregationHelper(BaseICDSAggregationHelper):
     helper_key = 'location'
     base_tablename = 'awc_location'
+    local_tablename = 'awc_location_local'
 
-    ucr_location_table = AWC_LOCATION_TABLE_ID
     ucr_aww_table = AWW_USER_TABLE_ID
 
-    local_tablename = 'awc_location_local'
-
     def __init__(self):
         pass
 
+    @property
+    def _table_column_names(self):
+        return (
+            'doc_id', 'awc_name', 'awc_site_code', 'awc_is_test',
+            'supervisor_id', 'supervisor_name', 'supervisor_site_code', 'supervisor_is_test',
+            'block_id', 'block_name', 'block_site_code', 'block_is_test', 'block_map_location_name',
+            'district_id', 'district_name', 'district_site_code', 'district_is_test', 'district_map_location_name',
+            'state_id', 'state_name', 'state_site_code', 'state_is_test', 'state_map_location_name',
+            'aggregation_level',
+        )
+
+    def generate_csv(self):
+        domain_locations = SQLLocation.objects.filter(domain=self.domain).values(
+            'pk', 'parent_id', 'metadata', 'name', 'location_id', 'location_type__code')
+        locations_by_pk = {
+            loc['pk']: loc
+            for loc in domain_locations
+        }
+        locations = [loc for loc in domain_locations if loc['location_type__code'] == 'awc']
+
+        output = io.StringIO()
+        writer = csv.DictWriter(
+            output, fieldnames=self._table_column_names,
+            delimiter='\t', escapechar='\\', lineterminator='\n'
+        )
+        writer.writeheader()
+
+        for location in locations:
+            metadata = json.loads(location['metadata'])
+            loc = {
+                'aggregation_level': 5,
+                'doc_id': location['location_id'],
+                'awc_name': location['name'].replace("\n", ""),
```

Comment:
some locations in `icds-dashboard-qa` have newlines in their names. Its currently unclear to me if that's expected, but I doubt it is. I went with this replace method because there were issues with the following `copy_from` to copy into postgres

I need to see if this is hte case in the real `icds-cas` domain